src/app/js/public/MenuItem.js

```diff
@@ -0,0 +1,297 @@
+                            'nav-item',
+                            css(styles.menuItem),
+                            {
+                                [css(styles.drawerActive)]:
+                                    searchDrawer === 'open',
+                            },
+                        )}
+                    >
+                        {icon}
+                        {label}
+                    </div>
+                )
+            );
+        case 'admin':
+            return (
+                role === 'admin' && (
+                    <a
+                        href="/admin"
+                        className={classnames('nav-item', css(styles.menuItem))}
+                    >
+                        {icon}
+                        {label}
+                    </a>
+                )
+            );
+        case 'sign-in':
+            return (
+                role === 'not logged' && (
+                    <Link
+                        to="/login"
+                        className={classnames('nav-item', css(styles.menuItem))}
+                    >
+                        {icon}
+                        {label}
+                    </Link>
+                )
+            );
+        case 'sign-out':
+            return (
+                role !== 'not logged' && (
+                    <Link
+                        to="/login"
+                        className={classnames('nav-item', css(styles.menuItem))}
+                        onClick={logout}
+                    >
+                        {icon}
+                        {label}
+                    </Link>
+                )
+            );
+
+        case 'custom': {
+            const { link } = config;
+            if (!link) {
+                console.error(
+                    `Missing link for custom menuItem: ${JSON.stringify(
+                        config,
+                    )}`,
+                );
+                return null;
+            }
+            if (link.startsWith('http')) {
+                return (
+                    <a
+                        href={link}
+                        className={classnames('nav-item', css(styles.menuItem))}
+                    >
+                        {icon}
+                        {label}
+                    </a>
+                );
+            }
+            return (
+                <NavLink
+                    to={link}
+                    onClick={closeAll}
+                    className={classnames(
+                        'nav-item',
+                        css(styles.menuItem),
+                        css(styles.link),
+                    )}
+                    activeClassName={css(styles.active)}
+                >
+                    {icon}
+                    {label}
+                </NavLink>
+            );
+        }
+        default:
+            console.error(
+                `Unknow role: ${config.role} menu item: ${JSON.stringify(
```

Comment:
You just have to use `custom` as the default, no ?
I would avoid to get an error message for nothing.